### PR TITLE
feat: #487 add modal to checkbox metrics

### PR DESCRIPTION
**Pull request by Alex89198900**
1.  Add modal to checkbox metrics

### CODE REVIEW

**Review by tritonJS826 — APPROVED** [review]

File: mw-migrations/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "migrate-goal-goalMetrics": "ts-node ./src/migrations/goalsMigration-add-goalMetrics.ts",
     "migrate-goal-delete": "ts-node ./src/migrations/goalMigration-delete-goalMetrics.ts",
     "export-backup": "ts-node ./src/backup/export.ts",
-    "import-backup": "ts-node ./src/backup/import.ts"
+    "import-backup": "node --loader ts-node/esm ./src/backup/import.ts"
```
[the remark this answers is not in the document]
Comment (Alex89198900) [reply mw-migrations/package.json:23]: Done

File: mw-migrations/backups/users.bkp.json [hunk 2]
```diff
@@ -0,0 +1 @@
...ngified":["{\"id\":\"70fa18fa-274a-402a-b91d-1d155e156a7e\",\"name\":\"New collection\",\"wayUuids\":[]}","{\"id\":\"5af26c62-7102-444b-9ef8-806d2b90dfd6\",\"name\":\"New collection\",\"wayUuids\":[]}"],"favoriteForUserUuids":[],"description":"UX/UI designer\n","wayRequestUuids":[],"imageUrl":"","tagsStringified":[],"uuid":"svHsdrnDJRe84en2GeZdwIBVbtr2"},{"name":"Виктория Умецкая","favoriteUserUuids":[],"wayRequestUuids":[],"favoriteWayUuids":[],"favoriteForUserUuids":[],"uuid":"twGBL0J5Q6Nr30uRS58omUrRt632","imageUrl":"","tagsStringified":[],"email":"vu1061994@gmail.com","customWayCollectionsStringified":[],"mentoringWayUuids":[],"ownWayUuids":["iPELPYQ67ve197g9nkyU"],"createdAt":{"seconds":1705578861,"nanoseconds":618000000},"description":"","isMentor":false},{"uuid":"uNHZ8bM0S7bywgKGkAAIBIgsu2k1","wayRequestUuids":[],"favoriteWayUuids":[],"mentoringWayUuids":[],"ownWayUuids":["F3y3Ko0Y0RNsK6QDgU8n"],"favoriteUserUuids":[],"tagsStringified":[],"isMentor":false,"createdAt":{"seconds":1705246108,"nanoseconds":398000000},"email":"viktoriabirukova935@gmail.com","description":"","favoriteForUserUuids":[],"name":"Viktoriia Byriukova","imageUrl":"","customWayCollectionsStringified":[]}]
```
[the remark this answers is not in the document]
Comment (Alex89198900) [reply mw-migrations/backups/users.bkp.json:1]: Done

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx [hunk 3]
```diff
@@ -44,19 +45,43 @@ export const GoalMetricItem = (props: SingleGoalMetricProps) => {
     ? `Done date ${DateUtils.getShortISODateValue(props.metric.doneDate)}`
     : "Not finished yet...";
 
+  /**
+   * Ghhgtt
```
[the remark this answers is not in the document]
Comment (Alex89198900) [reply mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx:49]: Done)

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx [hunk 4]
```diff
@@ -44,19 +45,43 @@ export const GoalMetricItem = (props: SingleGoalMetricProps) => {
...
-          className={styles.checkbox}
-          // TODO #487: check logic about update doneDate. What if I accidentally clicked checkbox few times
-          onChange={(isDone) => props.updateMetric({...props.metric, isDone, doneDate: new Date()})}
-        />
+        {!props.metric.isDone ?
+          <Checkbox
+            isEditable={props.isEditable}
+            isDefaultChecked={props.metric.isDone}
+            className={styles.checkbox}
+            // TODO #487: check logic about update doneDate. What if I accidentally clicked checkbox few times
```
[the remark this answers is not in the document]
Comment (Alex89198900) [reply mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx:66]: Done

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx [hunk 5]
```diff
@@ -44,19 +45,43 @@ export const GoalMetricItem = (props: SingleGoalMetricProps) => {
...
+            onChange={(isDone) => props.updateMetric({...props.metric, isDone, doneDate: new Date()})}
+          /> :
+          <Confirm
+            content={<p>
+              {"Are you sure "}
```
[the remark this answers is not in the document]
Comment (Alex89198900) [reply mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx:71]: Done

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx [hunk 6]
```diff
@@ -44,19 +45,41 @@ export const GoalMetricItem = (props: SingleGoalMetricProps) => {
...
+          /> :
+          <Confirm
+            content={<p>
+              {`Are you sure that you want set metric ${props.metric.description} not completed`}
+            </p>}
+            onOk={onOk}
+            okText="Confirm"
+            trigger={
+              <Checkbox
+                isEditable={false}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx:76]: What happened if not owner clicked on completed metric? Does modal appears?

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx [hunk 7]
```diff
@@ -44,19 +45,41 @@ export const GoalMetricItem = (props: SingleGoalMetricProps) => {
...
+                isDefaultChecked={props.metric.isDone}
+                className={styles.checkbox}
+                onChange={(isDone) => props.updateMetric({...props.metric, isDone, doneDate: new Date()})}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx:79]: Why do we need on change here if it is always not editable?

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx [hunk 8]
```diff
@@ -44,19 +45,41 @@ export const GoalMetricItem = (props: SingleGoalMetricProps) => {
...
+        {!props.metric.isDone ?
+          <Checkbox
+            isEditable={props.isEditable}
+            isDefaultChecked={props.metric.isDone}
+            className={styles.checkbox}
+            onChange={(isDone) => props.updateMetric({...props.metric, isDone, doneDate: new Date()})}
+          /> :
+          <Confirm
+            content={<p>
+              {`Are you sure that you want set metric ${props.metric.description} not completed`}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx:70]:
```suggestion
              {`Are you sure that you want set metric ${props.metric.description} as not completed`}
```

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx [hunk 9]
```diff
@@ -44,19 +45,41 @@ export const GoalMetricItem = (props: SingleGoalMetricProps) => {
...
-          onChange={(isDone) => props.updateMetric({...props.metric, isDone, doneDate: new Date()})}
-        />
+        {!props.metric.isDone ?
+          <Checkbox
+            isEditable={props.isEditable}
+            isDefaultChecked={props.metric.isDone}
+            className={styles.checkbox}
+            onChange={(isDone) => props.updateMetric({...props.metric, isDone, doneDate: new Date()})}
+          /> :
+          <Confirm
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx:68]: Does it possible to render only have one checkbox with nu ternnarian operator? Will it look better?

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx [hunk 10]
```diff
@@ -1,3 +1,4 @@
+
```
[the remark this answers is not in the document]
Comment (Alex89198900) [reply mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx]: Done

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx (hunk 6 above)
Comment (Alex89198900) [reply mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx:69]: Good point. There was a problem. But now it's fixed.

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx (hunk 7 above)
Comment (Alex89198900) [reply mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx]: Done

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx (hunk 9 above)
Comment (Alex89198900) [reply mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx:61]: I think in this case this is the most suitable solution if you do not change the modal

File: mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx [hunk 11]
```diff
@@ -44,19 +44,41 @@ export const GoalMetricItem = (props: SingleGoalMetricProps) => {
     ? `Done date ${DateUtils.getShortISODateValue(props.metric.doneDate)}`
     : "Not finished yet...";
 
+  /**
+   * Set metric not completed
+   */
+  const onOk = () => {
+    props.updateMetric({...props.metric, isDone: !props.metric.isDone, doneDate: new Date()});
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/goalMetricsBlock/GoalMetricItem.tsx:51]: by name looks like this function should not toggle the "isDone" field, but set it to false. 

```suggestion
    props.updateMetric({...props.metric, isDone: false, doneDate: new Date()});
```

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**